src/main/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactory.java

```diff
@@ -95,8 +96,12 @@ public CsvTransactionFactory(ObjectMapper objectMapper) {
                 result.put(FIELD_GOVUK_PAYMENT_ID, transactionEntity.getExternalId());
                 result.put(FIELD_AMOUNT, penceToCurrency(transactionEntity.getAmount()));
                 result.put(FIELD_TOTAL_AMOUNT, penceToCurrency(totalAmount));
-                result.put(FIELD_NET, penceToCurrency(netAmount));
                 result.put(FIELD_FEE, penceToCurrency(transactionEntity.getFee()));
+                if (nonNull(transactionEntity.getFee())) {
```

Comment:
We only need to set net_amount to `0` for failed transactions.

Fee is only applicable for stripe payments. So for worldpay payments that are successful this will result in net_amount of `0` for the csv.

Reply:
I guess for any payment that doesn't have a `net_amount` set on the transaction, we should default to 0. We also want to return 0 for in progress payments too I think, not just failed payments.

Services would only see the Net column for Worldpay payments in the CSV if they have a mix of Worldpay and Stripe services, and download from the "View transactions for all services" screen. Or I guess if they've switched provider from Worldpay to Stripe (or the other way round) and are looking at historic transactions.
Do we want to check the payment provider in here to only return 0 for Stripe, otherwise, return the amount?

Reply:
Updated to match the above requirements.